deltacat/compute/compactor/compaction_session.py

```diff
@@ -194,79 +167,83 @@ def _execute_compaction_round(
-                    f"partition locator: {rebase_source_partition_locator}")
-
-    # use the new hash bucket count if provided, or fall back to old count
-    hash_bucket_count = new_hash_bucket_count \
-        if new_hash_bucket_count is not None \
-        else old_hash_bucket_count
-
     # discover input delta files
+    # For rebase:
+    # Copy the old compacted table to a new destination, plus any new deltas from rebased source
+
+    # For incremental compaction:
+    # Input deltas from two sources: One: new delta; Two: compacted table
+
     high_watermark = round_completion_info.high_watermark \
         if round_completion_info else None
 
+    # Source One: new deltas from uncompacted table for incremental compaction, or deltas from compacted table for rebase
     input_deltas = io.discover_deltas(
         source_partition_locator,
-        high_watermark,
-        last_stream_position_to_compact,
+        high_watermark[source_partition_locator.canonical_string()] if isinstance(high_watermark,
+                                                                                  dict) else high_watermark,
+        last_stream_position_to_compact if not rebase_source_partition_locator else deltacat_storage.get_partition(
+            source_partition_locator.stream_locator,
+            source_partition_locator.partition_values).stream_position,
         deltacat_storage,
     )
 
+    # Source Two: compacted table in case of incremental compaction or new deltas from uncompacted table
+    if not rebase_source_partition_locator:  # compacted table
+        compacted_partition = deltacat_storage.get_partition(compacted_partition_locator.stream_locator,
+                                                                        compacted_partition_locator.partition_values)
+        compacted_last_stream_position = compacted_partition.stream_position if compacted_partition else None
+        input_deltas_compacted = []
+        if compacted_last_stream_position:
+            input_deltas_compacted = io.discover_deltas(
+                compacted_partition_locator,
+                None,
+                compacted_last_stream_position,
+                deltacat_storage
+            )
+        logger.info(
+            f"Length of input deltas from uncompacted table {len(input_deltas)} up to {last_stream_position_to_compact},"
+            f"Length of input deltas from compacted table {len(input_deltas_compacted)} up to {high_watermark}")
+        input_deltas += input_deltas_compacted
+    else:  # new deltas from uncompacted table based on inferred last stream position to last position to compact
+        input_deltas_new = io.discover_deltas(
+            rebase_source_partition_locator,
+            rebase_source_partition_high_watermark if rebase_source_partition_high_watermark else io.getLastCompactedDeltaStreamPosition(
+                source_partition_locator,
+                deltacat_storage
+            ),
```

Comment:
Minor: rename to `get_last_compacted_delta_stream_position`, and probably move this defaulting logic to **psuedo private** method starting with `_` (underscore).